server/src/main/java/org/elasticsearch/snapshots/SnapshotShardsService.java

```diff
@@ -328,7 +328,11 @@ private void processIndexShardSnapshots(ClusterChangedEvent event) {
 
                 for (final Map.Entry<ShardId, IndexShardSnapshotStatus> shardEntry : entry.getValue().entrySet()) {
                     final ShardId shardId = shardEntry.getKey();
-                    final IndexShard indexShard = indicesService.indexServiceSafe(shardId.getIndex()).getShardOrNull(shardId.id());
+                    final IndexShard indexShard = indicesService.getShardOrNull(shardId);
+                    if (indexShard == null) {
+                        notifyFailedSnapshotShard(snapshot, shardId, localNodeId, "could not find shard");
```

Comment:
Is this change necessary to fix the test failure? Should we do this as part of the follow-up refactoring of SnapshotShardsService?

Reply:
@ywelsch I was able to reproduce running into this one with a very low frequency running the newly added test (though I just tried it again for an hour in a loop and couldn't so maybe this isn't an issue anymore ... I'll revert it for now, technically this shouldn't be necessary).